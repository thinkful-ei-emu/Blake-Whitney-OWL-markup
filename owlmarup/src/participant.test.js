import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Participant from './participant';

describe('<Participant />', () => {
  // Smoke Test
  it('renders without crashing', () => {
    // Create a DOM element to render the component into
    const div = document.createElement('div');

    // Render the component
    // If something is wrong it will fail here
    ReactDOM.render(<Participant person={{
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  }}/>, div);

    // Clean up
    ReactDOM.unmountComponentAtNode(div);
  });

  // Snapshot Test
  it('renders this UI as expected', () => {
    const tree = renderer.create(<Participant key={1} person={{
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  }}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});