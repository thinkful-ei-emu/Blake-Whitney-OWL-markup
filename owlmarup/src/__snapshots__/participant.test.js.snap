// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Participant /> renders this UI as expected 1`] = `
<div
  className="participant"
>
  <img
    alt="person avatar"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <h2
    className="name"
  >
    Koren Templeton
  </h2>
  <p
    className="onStage"
  >
    On Stage
  </p>
</div>
`;
